openlibrary/catalog/marc/marc_binary.py

```diff
@@ -106,14 +106,14 @@ def get_lower_subfields(self):
 class MarcBinary(MarcBase):
     def __init__(self, data):
         try:
-            assert len(data) and isinstance(data, six.string_types)
+            assert len(data) and isinstance(data, bytes)
             length = int(data[:5])
         except:
             raise BadMARC("No MARC data found")
```

Comment:
@cclauss I think this is good, but I think the full exception should not go in the response since it will be returned over a JSON API. I think it is ok to return a short msg that nothing resembling MARC data was found, but I'll review it once the tests are passing.

Reply:
There's enough going on in this PR now, just want to get the type sorted and existing tests passing. This can come in a separate PR.

Reply:
I have made a change to this code, to avoid catching things inheriting directly from `BaseException` , which is not desired here. That links is complaining specifically about `except: / pass` code, which I have seen in OL and it is not good. This is slightly different in that the intent is to catch something unexpected, and it immediately raises another exception at the outer layer of our API interface to indicate that something fundamentally went wrong, and no expected MARC record could be found.  Some of the more nuanced responses here https://stackoverflow.com/questions/4990718/about-catching-any-exception helped clarify my view that this isn't the worst version of this pattern, and it is at an appropriate level, and the code (at lesat now) reflects the intent.   Missing an unusual Error type in a full enumerated list and not raising a `BadMARC` error would be a regression, and worse.